libs/core/src/main/java/org/opensearch/core/xcontent/filtering/FilterPath.java

```diff
@@ -99,32 +99,26 @@ public static FilterPath[] compile(Set<String> filters) {
 
         List<FilterPath> paths = new ArrayList<>();
         for (String filter : filters) {
-            if (filter != null) {
+            if (filter != null && !filter.isEmpty()) {
                 filter = filter.trim();
                 if (filter.length() > 0) {
-                    paths.add(parse(filter, filter));
+                    paths.add(parse(filter));
                 }
             }
         }
         return paths.toArray(new FilterPath[0]);
     }
 
-    private static FilterPath parse(final String filter, final String segment) {
-        int end = segment.length();
+    private static FilterPath parse(final String filter) {
+        String[] segments = filter.split("(?<!\\\\)\\.");
```

Comment:
nit: Could add comments her for better readability? 
Eg: // Split the filter into segments using a regex that avoids splitting escaped dots.